### ISSUE TITLE
Add a new docs section to explain models evolution compatibility

### ISSUE THREAD

**Issue by rafaparadela** [frees-io/freestyle-rpc#335]
It would be helpful to explain how models can evolve, and how the serializers preserve back/forward compatibility.

As part of this new section, we can add [these conclusions for the Avro](https://github.com/frees-io/freestyle-rpc/issues/331#issuecomment-401789947) cases, but we probably should do the same thing for Protocol Buffer definitions.

**Comment by fedefernandez** [comment]
@juanpedromoreno this is done, right?

**Comment by juanpedromoreno** [comment]
Yes, it's partially done since we might want to add the proto section.
